src/pcapi/core/users/repository.py

```diff
@@ -31,12 +32,20 @@
 logger = logging.getLogger(__name__)
 
 
+hashed_placeholder = crypto.hash_password("placeholder")
+
+
 def check_user_and_credentials(user: User, password: str) -> None:
     # Order is important to prevent end-user to guess user emails
     # We need to check email and password before checking email validation
-    if not user or not user.isActive or not user.checkPassword(password):
-        if user:
-            logging.info("Failed authentication attempt", extra={"user": user.id, "avoid_current_user": True})
+    if not user:
+        # Hash the given password, just like we would do if the user
+        # existed. This avoid user enumeration by comparing server
+        # response time.
+        crypto.check_password(password, hashed_placeholder)
+        raise exceptions.InvalidIdentifier()
+    if not user.isActive or not user.checkPassword(password):
```

Comment:
du coup, tant qu'on y est:
```suggestion
    if not user.checkPassword(password) or not user.isActive:
```

Reply:
oui, bien vu.